pandas/_libs/src/ujson/python/date_conversions.c

```diff
@@ -93,10 +93,14 @@ char *PyDateTimeToIso(PyObject *obj, NPY_DATETIMEUNIT base,
     // Check to see if PyDateTime has a timezone.
     // Don't convert to UTC if it doesn't.
     int is_tz_aware = 0;
-    PyObject *offset = extract_utc_offset(obj);
-    if (offset != NULL) {
-        is_tz_aware = 1;
-        Py_DECREF(offset);
+    if (PyObject_HasAttrString(obj, "tzinfo")) {
+        PyObject *offset = extract_utc_offset(obj);
+        if (offset != NULL) {
```

Comment:
I think if you get NULL back from `extract_utc_offset` you still want to propogate up the stack, so would be better to do something like:

```
if (offset == NULL) {
  return NULL;
}

is_tz_aware = offset != Py_None;
...
...
```